src/diceBot/DoubleCross.rb

```diff
@@ -40,9 +40,9 @@ def getHelpMessage
   #
   # 1. ダイス数
   # 2. 修正値
-  # 3. クリティカル値
-  # 4. 達成値
-  DX_OD_TOLL_RE = /\A(\d+)DX([-+][-+\d]+)?@(\d+)(?:>=(\d+))?\z/io.freeze
+  # 6. クリティカル値
+  # 7. 達成値
+  DX_OD_TOOL_RE = /\A(\d+)DX(([-+](\d+|\d+[\+\-\*]\d+|\(\d+([\+\-\*]\d+)*\)))+)?@(\d+)(?:>=(\d+))?\z/io.freeze
```

Comment:
```suggestion
  DX_OD_TOOL_RE = /\A(\d+)DX([-+][-+*\d]+)?@(\d+)(?:>=(\d+))?\z/io.freeze
```

`()`の計算は事前に処理されるのでここに含める必要はありません